components/builder-api/src/server/resources/profile.rs

```diff
@@ -97,6 +103,18 @@ fn generate_access_token(req: HttpRequest<AppState>) -> HttpResponse {
         session.get_flags()
     };
 
+    // Memcache supports multiple tokens but to preserve legacy behavior
+    // we must purge any existing tokens before generating new ones
+    match do_get_access_tokens(&req, account_id) {
+        Ok(access_tokens) => {
+            let mut memcache = req.state().memcache.borrow_mut();
+            for token in access_tokens.get_tokens() {
+                memcache.delete_key(token.get_token())
```

Comment:
I wonder if there is a race condition here.. the time between we delete from cache and we delete from DB.  We may want to move this AFTER we delete from DB.